internal/operator/terraform/operator.go

```diff
@@ -0,0 +1,207 @@
+func (t *Terraform) Create(p types.ProviderType, cfg map[string]interface{}) (*types.ClusterInfo, error) {
+	// silence stdErr during terraform execution, plugins send debug and trace entries there
+	stderr := os.Stderr
+	os.Stderr, _ = os.Open(os.DevNull)
+	defer func() { os.Stderr = stderr }()
+
+	// init cluster files
+	if !t.ops.Persistent {
+		// remove all files if not persistent after running
+		defer cleanup(t.ops.DataDir(), cfg["project"].(string), cfg["cluster_name"].(string), p)
+	}
+	if err := initClusterFiles(t.ops.DataDir(), p, cfg); err != nil {
+		return nil, errors.Wrap(err, "Could not initialize cluster data")
+	}
+
+	clusterDir, err := clusterDir(t.ops.DataDir(), cfg["project"].(string), cfg["cluster_name"].(string), p)
+	if err != nil {
+		return nil, err
+	}
+
+	// INIT
+	if p == types.Gardener {
+		if err := initGardenerProvider(); err != nil {
+			return nil, errors.Wrap(err, "could not initialize the gardener provider")
+		}
+	}
+	i := &command.InitCommand{
+		Meta: t.ops.Meta,
+	}
+
+	e := i.Run(t.initArgs(p, cfg, clusterDir))
+	if e != 0 {
+		return nil, t.checkUIErrors()
+	}
+
+	// APPLY
+	a := &command.ApplyCommand{
+		Meta: t.ops.Meta,
+	}
+	e = a.Run(t.applyArgs(p, cfg, clusterDir))
+	if e != 0 {
+		return nil, t.checkUIErrors()
+	}
+
+	return clusterInfoFromFile(t.ops.DataDir(), cfg["project"].(string), cfg["cluster_name"].(string), p)
+}
+
+// Status checks the current state of the cluster from the file
+func (t *Terraform) Status(sf *statefile.File, p types.ProviderType, cfg map[string]interface{}) (*types.ClusterStatus, error) {
+	cs := &types.ClusterStatus{
+		Phase: types.Unknown,
+	}
+	var err error
+
+	// if no state given, try the file system
+	if sf == nil {
+		sf, err = stateFromFile(t.ops.DataDir(), cfg["project"].(string), cfg["cluster_name"].(string), p)
+		if err != nil {
+			return cs, err
+		}
+	}
+
+	if sf.State.HasResources() {
+		cs.Phase = types.Provisioned
+	}
+
+	return cs, nil
+}
+
+// Delete removes an existing cluster or returns an error if removing the cluster is not possible.
+func (t *Terraform) Delete(sf *statefile.File, p types.ProviderType, cfg map[string]interface{}) error {
+	// silence stdErr during terraform execution, plugins send debug and trace entries there
+	stderr := os.Stderr
+	os.Stderr, _ = os.Open(os.DevNull)
+	defer func() { os.Stderr = stderr }()
+
+	// init cluster files
+	if !t.ops.Persistent {
+		// remove all files if not persistent after running
+		defer cleanup(t.ops.DataDir(), cfg["project"].(string), cfg["cluster_name"].(string), p)
+	}
+	if err := initClusterFiles(t.ops.DataDir(), p, cfg); err != nil {
+		return errors.Wrap(err, "Could not initialize cluster data")
+	}
+
+	var err error
+	// if no state given, check if it is already in the file system
+	if sf == nil {
+		_, err = stateFromFile(t.ops.DataDir(), cfg["project"].(string), cfg["cluster_name"].(string), p)
+		if err != nil {
+			return err
```

Comment:
In case of a permission problem, this error can be wrapped and enriched with the context of why we try to access to the file system.

Reply:
Done thanks!